extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/serialization/JsonProducer.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.kafka.client.serialization;
+
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
+
+final class JsonProducer {
+
+    private JsonProducer() {
+    }
+
+    // Try to get JSON-B from Arc but fallback to regular JSON-B creation
+    // The fallback could be used for example in unit tests where Arc has not been initialized
+    static Jsonb getJsonb() {
+        Jsonb jsonb = null;
+        ArcContainer container = Arc.container();
+        if (container != null) {
+            jsonb = container.instance(Jsonb.class).get();
+        }
+        return jsonb != null ? jsonb : JsonbBuilder.create();
+    }
+
+    // same comment as above but for Jackson's ObjectMapper
+    static ObjectMapper getObjectMapper() {
+        ObjectMapper objectMapper = null;
+        ArcContainer container = Arc.container();
+        if (container != null) {
+            objectMapper = container.instance(ObjectMapper.class).get();
+        }
+        return objectMapper != null ? objectMapper : new ObjectMapper();
+    }
```

Comment:
Won't you have an error if you only have one of the dependencies at runtime. I mean either `Jsonb` or `ObjectMapper` won't exist in the classpath?

Reply:
I'm waiting for CI to tell me that :)

Reply:
CI passed. I will however later on do some more manual tests and see how if there is anything that we might have missed.

Reply:
Thanks for catching this!

The PR has been updated to make sure this is not a problem.